tests/MongoDB.Driver.Core.Tests/Core/Configuration/ConnectionSettingsTests.cs

```diff
@@ -168,25 +184,28 @@ public void With_applicationName_should_return_expected_result()
             var result = subject.With(applicationName: newApplicationName);
 
             result.ApplicationName.Should().Be(newApplicationName);
-            result.Authenticators.Should().Equal(subject.Authenticators);
+            AssertActualAuthenticatorsFactoryResult(result.AuthenticatorFactories, subject.AuthenticatorFactories);
```

Comment:
Replace with:

            result.AuthenticatorFactories.Should().Equal(subject.AuthenticatorFactories);